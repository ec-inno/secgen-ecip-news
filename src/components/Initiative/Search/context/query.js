import React from 'react';

import { defaultLangKey } from '../../../../../languages';

const queryInit = {
  filters: {},
  status: 'ALL',
  language: defaultLangKey,
  pagination: '0/10',
};

const queryReducer = (state, action) => {
  switch (action.type) {
    case 'paginate': {
      return {
        ...state,
        pagination: action.pagination,
      };
    }

    case 'changeStatus': {
      return {
        ...state,
        pagination: queryInit.pagination,
        status: action.status,
      };
    }

    case 'changeLanguage': {
      return {
        ...state,
        pagination: queryInit.pagination,
        language: action.language,
      };
    }

    case 'changeFilter': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          [action.filter]: action.filterValue,
        },
      };
    }

    case 'setFilters': {
      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...state.filters,
          ...action.filters,
        },
      };
    }

    case 'removeFilter': {
      const { filters } = state;
      delete filters[action.filter];

      return {
        ...state,
        pagination: queryInit.pagination,
        filters: {
          ...filters,
        },
      };
    }

    case 'reset':
    default: {
      return queryInit;
    }
  }
};

const Context = React.createContext();

export { queryInit, queryReducer, Context as default };
