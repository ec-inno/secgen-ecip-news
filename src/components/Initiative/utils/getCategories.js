const getCategories = t => {
  return [
    { value: 'any', label: t('All categories') },
    { value: 'AGRI', label: t('Agriculture') },
    { value: 'DEVCO', label: t('Aid and development cooperation') },
    { value: 'EURO', label: t('Business and economy') },
    { value: 'SANTE', label: t('Consumers and health') },
    { value: 'CULT', label: t('Culture and media') },
    { value: 'DECO', label: t('Digital economy and society') },
    { value: 'EDU', label: t('Education, youth and sport') },
    { value: 'EMPL', label: t('Employment and social affairs') },
    { value: 'ENER', label: t('Energy') },
    { value: 'ENV', label: t('Environment and climate') },
    { value: 'TRADE', label: t('External trade and relations') },
    { value: 'JUST', label: t('Justice and fundamental rights') },
    { value: 'MARE', label: t('Maritime affairs and fisheries') },
    { value: 'MIGR', label: t('Migration and asylum') },
    { value: 'REGIO', label: t('Regional development') },
    { value: 'RSH', label: t('Research') },
    { value: 'SEC', label: t('Security') },
    { value: 'TRA', label: t('Transport') },
    { value: 'REGFRA', label: t('REGFRA') },
    { value: 'REGECI', label: t('REGECI') },
    { value: 'COLLECI', label: t('COLLECI') },
    { value: 'EXAECI', label: t('EXAECI') },
  ];
};

export default getCategories;
