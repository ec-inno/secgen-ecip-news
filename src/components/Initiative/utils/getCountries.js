const getCountriesTranslated = t => {
  return [
    { value: 'bg', label: t('Bulgaria') },
    { value: 'es', label: t('Spain') },
    { value: 'cs', label: t('Czech Republic') },
    { value: 'da', label: t('Denmark') },
    { value: 'de', label: t('Germany') },
    { value: 'et', label: t('Estonia') },
    { value: 'el', label: t('Greece') },
    { value: 'fr', label: t('France') },
    { value: 'ga', label: t('Ireland') },
    { value: 'hr', label: t('Croatia') },
    { value: 'it', label: t('Italia') },
    { value: 'lv', label: t('Latvia') },
    { value: 'lt', label: t('Lithuania') },
    { value: 'hu', label: t('Hungary') },
    { value: 'mt', label: t('Malta') },
    { value: 'nl', label: t('Netherlands') },
    { value: 'pl', label: t('Poland') },
    { value: 'pt', label: t('Portugal') },
    { value: 'ro', label: t('Romania') },
    { value: 'sk', label: t('Slovakia') },
    { value: 'sl', label: t('Slovenia') },
    { value: 'fi', label: t('Finland') },
    { value: 'sv', label: t('Sweden') },
  ];
};

export default getCountriesTranslated;
