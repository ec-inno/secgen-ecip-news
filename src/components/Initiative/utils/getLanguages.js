const getLanguagesTranslated = t => {
  return [
    { value: 'bg', label: t('Bulgarian') },
    { value: 'es', label: t('Spanish') },
    { value: 'cs', label: t('Czech') },
    { value: 'da', label: t('Dansk') },
    { value: 'de', label: t('German') },
    { value: 'et', label: t('Estonian') },
    { value: 'el', label: t('Greek') },
    { value: 'en', label: t('English') },
    { value: 'fr', label: t('French') },
    { value: 'ga', label: t('Irish') },
    { value: 'hr', label: t('Croatian') },
    { value: 'it', label: t('Italian') },
    { value: 'lv', label: t('Latvian') },
    { value: 'lt', label: t('Lithuanian') },
    { value: 'hu', label: t('Hungarian') },
    { value: 'mt', label: t('Maltese') },
    { value: 'nl', label: t('Dutch') },
    { value: 'pl', label: t('Polish') },
    { value: 'pt', label: t('Portuguese') },
    { value: 'ro', label: t('Romanian') },
    { value: 'sk', label: t('Slovak') },
    { value: 'sl', label: t('Slovene') },
    { value: 'fi', label: t('Finnish') },
    { value: 'sv', label: t('Swedish') },
  ];
};

export default getLanguagesTranslated;
