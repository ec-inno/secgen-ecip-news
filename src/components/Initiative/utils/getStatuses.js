// These are values for filters, not resource path!
// Resource path values are LATEST, ONGOING, ANSWERED, ALL, REFUSED
const getStatuses = t => {
  return [
    { value: 'ALL', label: t('All') },
    { value: 'OPEN', label: t('Open') },
    { value: 'SUCCESSFUL', label: t('Successful') },
    { value: 'ARCHIVED', label: t('Archived') },
  ];
};

export default getStatuses;
