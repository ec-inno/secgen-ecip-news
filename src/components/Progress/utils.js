export const steps = [
  'REGISTERED',
  'COLLECTION_START_DATE',
  'ONGOING',
  'COLLECTION_EARLY_CLOSURE',
  'CLOSED',
  'VERIFICATION',
  'SUBMITTED',
  'ANSWERED',
];

/**
 * Reorder progress stages to match order of `steps`.
 * @param {Array<Object>} progress
 * @param {String} progress.name
 * @param {Boolean} progress.active
 * @param {Date} progress.date
 */
export const getStages = progress => {
  const stages = [];

  steps.forEach(step => {
    const match = progress.find(item => item.name === step);
    if (match) stages.push(match);
  });

  return stages;
};
