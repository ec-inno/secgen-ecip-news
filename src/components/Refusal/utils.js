import messages from './messages.json';

export const reasonsSupported = [
  'reason.action.registration.reject.abuse',
  'reason.action.registration.reject.competences',
  'reason.action.registration.reject.requirements',
  'reason.action.registration.reject.values',
];

export const getTranslation = ({ code, language }) =>
  messages.find(
    message => message.LNG_CODE === language && message.RTR_KEY === code
  );

export const getReasons = ({ input, language }) => {
  const output = [];

  input.forEach(reason => {
    if (reasonsSupported.includes(reason)) {
      const translation = getTranslation({ code: reason, language });
      if (translation) {
        output.push(translation);
      }
    }
  });

  return output;
};
