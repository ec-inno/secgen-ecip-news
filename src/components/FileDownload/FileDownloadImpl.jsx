import React from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import mime from 'mime';

import { useI18nContext } from '@eci/context/I18n';
import formatBytes from '@eci/utils/formatBytes';
import { useDocumentApi, getResource } from '@eci/utils/useDocumentApi';
import getLanguages from '@eci/utils/getLanguages';

import Link from '../Link/LinkGatsby';
import ErrorMessage from '../ErrorMessage';
import { FileDownload } from './FileDownload';

const FileDownloadImpl = ({ file }) => {
  if (!file || Object.keys(file).length === 0) return '';
  if (!file.id && (!file.otherLanguages || file.otherLanguages.length === 0))
    return '';

  const { location } = useI18nContext();
  const { t } = useTranslation();
  const langsT = getLanguages(t);

  const { otherLanguages, language: fileLanguage } = file;

  const langT = langsT.find(l => l.value === fileLanguage);
  const language = langT ? langT.label : '';

  const languages = otherLanguages
    ? otherLanguages.map(language => {
        let urlPath = '';
        let pathParts = [];

        const translation = langsT.find(l => l.value === language);

        pathParts = location.pathname.split('/').filter(p => p);

        pathParts.shift();

        urlPath = pathParts.join('/');
        urlPath = `/${translation.value}/${urlPath}/`;

        if (location.hash) {
          urlPath += location.hash;
        }

        // Link attributes for the component's spread.
        return { label: translation.label, href: urlPath };
      })
    : [];

  if (!file.id && languages.length) {
    return (
      <p className="ecl-u-type-paragraph">
        {t('Other languages available')}:{' '}
        {languages.map((attr, key) => [
          key > 0 && ', ',
          <Link {...attr} key={key} />,
        ])}
      </p>
    );
  }

  const { error } = useDocumentApi(file.id);
  const resource = getResource(file.id);

  let meta = '';

  const title = file.name ? file.name : '';

  if (file.size) {
    meta += formatBytes(file.size);
  }

  if (file.mimeType) {
    meta += ' - ';
    meta += mime.getExtension(file.mimeType).toUpperCase();
  }

  if (Object.keys(error).length !== 0) {
    return <ErrorMessage title={t('File service error')} error={error} />;
  }

  return (
    <FileDownload
      title={title}
      language={language}
      meta={meta}
      download={{ label: t('Download'), href: resource }}
      icon={{ shape: 'general--copy', size: '2xl' }}
    />
  );
};

FileDownloadImpl.propTypes = {
  file: PropTypes.shape({
    name: PropTypes.string,
    language: PropTypes.string,
    otherLanguages: PropTypes.arrayOf(PropTypes.string),
    mimeType: PropTypes.string,
    id: PropTypes.number,
    size: PropTypes.number,
  }),
};

export default FileDownloadImpl;
