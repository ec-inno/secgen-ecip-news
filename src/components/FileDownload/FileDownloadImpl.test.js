import React from 'react';
import { render } from '@testing-library/react';

import * as useDocumentApi from '@eci/utils/useDocumentApi';
import * as I18nContext from '@eci/context/I18n';
import FileDownloadImpl from './FileDownloadImpl';

describe('FileDownloadImpl', () => {
  it('does not render on missing input', () => {
    const { container } = render(<FileDownloadImpl />);
    expect(container).toMatchSnapshot();
  });

  it('does not render anything on missing id', () => {
    const { container } = render(<FileDownloadImpl file={{}} />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly provided an input', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
      location: {
        href: 'http://localhost:8000/en/initiatives/#1',
        pathname: '/en/initiatives/',
        hash: '#1',
      },
    }));

    const props = {
      file: {
        name: 'test',
        mimeType: 'text/plain',
        id: 16806,
        size: 10,
      },
    };

    const { container } = render(<FileDownloadImpl {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('renders file extension only if mimeType is actually useful', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
      location: {
        href: 'http://localhost:8000/en/initiatives/#1',
        pathname: '/en/initiatives/',
        hash: '#1',
      },
    }));

    const props = {
      file: {
        name: 'test',
        mimeType: 'unknown',
        id: 16806,
        size: 10,
      },
    };

    const { container } = render(<FileDownloadImpl {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('renders language label when provided', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
      location: {
        href: 'http://localhost:8000/en/initiatives/#1',
        pathname: '/en/initiatives/',
        hash: '#1',
      },
    }));

    const props = {
      file: {
        language: 'en',
        name: 'test',
        mimeType: 'text/plain',
        id: 16806,
        size: 10,
      },
    };

    const { container } = render(<FileDownloadImpl {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('can render a message for alternative language versions when a file is not available', () => {
    jest.spyOn(I18nContext, 'useI18nContext').mockImplementation(() => ({
      locale: 'en',
      location: {
        href: 'http://localhost:8000/en/initiatives/#1',
        pathname: '/en/initiatives/',
        hash: '#1',
      },
    }));

    const props = {
      file: {
        otherLanguages: ['bg', 'cs'],
      },
    };

    const { container } = render(<FileDownloadImpl {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('renders an error message when file is not reachable', () => {
    jest.spyOn(useDocumentApi, 'useDocumentApi').mockImplementation(() => ({
      response: '',
      isLoading: false,
      error: { message: 'File not found.' },
    }));

    const props = {
      file: {
        name: 'test',
        mimeType: 'text/plain',
        id: 16806,
        size: 10,
      },
    };

    const { container } = render(<FileDownloadImpl {...props} />);
    expect(container).toMatchSnapshot();
  });
});
