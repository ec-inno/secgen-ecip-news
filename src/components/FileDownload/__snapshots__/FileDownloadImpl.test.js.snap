// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileDownloadImpl can render a message for alternative language versions when a file is not available 1`] = `
<div>
  <p
    class="ecl-u-type-paragraph"
  >
    Other languages available
    :
     
    <a
      class="ecl-link"
      href="/bg/initiatives/#1"
    >
      Bulgarian
    </a>
    , 
    <a
      class="ecl-link"
      href="/cs/initiatives/#1"
    >
      Czech
    </a>
  </p>
</div>
`;

exports[`FileDownloadImpl does not render anything on missing id 1`] = `
<div>
  
</div>
`;

exports[`FileDownloadImpl does not render on missing input 1`] = `
<div>
  
</div>
`;

exports[`FileDownloadImpl renders an error message when file is not reachable 1`] = `
<div>
  <div
    class="ecl-message ecl-message--error"
    data-ecl-message="true"
    role="alert"
  >
    <svg
      aria-hidden="true"
      class="ecl-message__icon ecl-icon ecl-icon--l"
      focusable="false"
    >
      
      
      <use
        xlink:href="file#notifications--error"
      />
    </svg>
    <div
      class="ecl-message__content"
    >
      <button
        class="ecl-message__close ecl-button ecl-button--ghost"
        data-ecl-message-close="true"
        type="button"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label="true"
          >
            Close
          </span>
          <svg
            aria-hidden="true"
            class="ecl-button__icon ecl-button__icon--after ecl-icon ecl-icon--s"
            data-ecl-icon="true"
            focusable="false"
          >
            
            
            <use
              xlink:href="file#ui--close"
            />
          </svg>
        </span>
      </button>
      <div
        class="ecl-message__title"
      >
        File service error
      </div>
      <p
        class="ecl-message__description"
      >
        File not found.
      </p>
    </div>
  </div>
</div>
`;

exports[`FileDownloadImpl renders correctly provided an input 1`] = `
<div>
  <div
    class="ecl-file"
    data-ecl-file="true"
  >
    <div
      class="ecl-file__container"
    >
      <svg
        aria-hidden="true"
        class="ecl-file__icon ecl-icon ecl-icon--2xl"
        focusable="false"
      >
        
        
        <use
          xlink:href="file#general--copy"
        />
      </svg>
      <div
        class="ecl-file__info"
      >
        <div
          class="ecl-file__title"
        >
          test
        </div>
        <div
          class="ecl-file__language"
        />
        <div
          class="ecl-file__meta"
        >
          10 Bytes - TXT
        </div>
      </div>
      <a
        class="ecl-file__download ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after"
        download=""
        href="undefined/register/document/16806"
      >
        <span
          class="ecl-link__label"
        >
          Download
        </span>
         
        <svg
          aria-hidden="true"
          class="ecl-link__icon ecl-icon ecl-icon--fluid"
          focusable="false"
        >
          
          
          <use
            xlink:href="file#ui--download"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`FileDownloadImpl renders file extension only if mimeType is actually useful 1`] = `
<div>
  <div
    class="ecl-file"
    data-ecl-file="true"
  >
    <div
      class="ecl-file__container"
    >
      <svg
        aria-hidden="true"
        class="ecl-file__icon ecl-icon ecl-icon--2xl"
        focusable="false"
      >
        
        
        <use
          xlink:href="file#general--copy"
        />
      </svg>
      <div
        class="ecl-file__info"
      >
        <div
          class="ecl-file__title"
        >
          test
        </div>
        <div
          class="ecl-file__language"
        />
        <div
          class="ecl-file__meta"
        >
          10 Bytes
        </div>
      </div>
      <a
        class="ecl-file__download ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after"
        download=""
        href="undefined/register/document/16806"
      >
        <span
          class="ecl-link__label"
        >
          Download
        </span>
         
        <svg
          aria-hidden="true"
          class="ecl-link__icon ecl-icon ecl-icon--fluid"
          focusable="false"
        >
          
          
          <use
            xlink:href="file#ui--download"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`FileDownloadImpl renders language label when provided 1`] = `
<div>
  <div
    class="ecl-file"
    data-ecl-file="true"
  >
    <div
      class="ecl-file__container"
    >
      <svg
        aria-hidden="true"
        class="ecl-file__icon ecl-icon ecl-icon--2xl"
        focusable="false"
      >
        
        
        <use
          xlink:href="file#general--copy"
        />
      </svg>
      <div
        class="ecl-file__info"
      >
        <div
          class="ecl-file__title"
        >
          test
        </div>
        <div
          class="ecl-file__language"
        >
          English
        </div>
        <div
          class="ecl-file__meta"
        >
          10 Bytes - TXT
        </div>
      </div>
      <a
        class="ecl-file__download ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after"
        download=""
        href="undefined/register/document/16806"
      >
        <span
          class="ecl-link__label"
        >
          Download
        </span>
         
        <svg
          aria-hidden="true"
          class="ecl-link__icon ecl-icon ecl-icon--fluid"
          focusable="false"
        >
          
          
          <use
            xlink:href="file#ui--download"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
